### ISSUE TITLE
[v3] Implement new Concept Exercise: `dynamic-dispatch`

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/elixir#558]
## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

## Goal

The goal of this exercise is to replace `rpn-calculator-output` as the exercise that teaches dynamic dispatch (https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#dynamic-dispatch). `rpn-calculator-output` should be modified to depend on `dynamic-dispatch` instead of teaching it. Make sure to reuse part of that exercise's introduction.

## Learning objectives

- Know that you module names are just atoms
- Know that you can call a function on a module even if its stored in a variable
- `apply`

## Out of scope

?

## Concepts

- `dynamic-dispatch`

## Prerequisites

Depends on the story.

## Resources to refer to

https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#dynamic-dispatch

## Implementing

Since dynamic dispatch depends on the existence of a few modules with the same function, we might want to provide that in the boilerplate instead of having the student implement those too.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by apoorv-2204** [comment]
Hi is issue still open?

**Comment by angelikatyborska** [comment]
@apoorv-2204 Yes, nobody started working on this yet.

**Comment by zxdsc** [comment]
Hello!

The link under resources leading not to where it expected. Is there still a thing in Elixir such as `dynamic-dispatch`?

**Comment by angelikatyborska** [comment]
@zxdsc This link https://hexdocs.pm/elixir/Kernel.html#apply/3 ? Yes, that's exactly the function that allows you to do dynamic dispatch in Elixir
